### PR TITLE
Feature/add indexed db support

### PR DESCRIPTION
**Pull request by giacomocerre**

### CODE REVIEW

File: src/app/services/idb.service.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    return this.db.add(data);
+  }
+
+  update(id, data) {
+    return this.db.update(id, data);
+  }
+
+  remove(id) {
+    return this.db.delete(id);
+  }
```
Comment (szenzaro) [inline src/app/services/idb.service.ts:33]: Why do we need to wrap db?
Do you expect the use of Dexie to change over time?

File: src/app/models/evt-models.ts [hunk 2]
```diff
@@ -263,3 +263,7 @@ export class Choice extends GenericElement {
     originalContent: Array<ParseResult<GenericElement>>;
     normalizedContent: Array<ParseResult<GenericElement>>;
 }
+
+export interface AnnotationID  {
+    annotationID: number;
+}
```
Comment (szenzaro) [inline src/app/models/evt-models.ts:269]: You can put this inside idb service

File: src/app/app-idb.module.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  providers: [
+    IdbService,
+  ],
+  bootstrap: [],
+})
+export class AppIdbCoreModule {
+  constructor(@Optional() @SkipSelf() parentModule: AppIdbCoreModule) {
+    if (parentModule) {
+      throw new Error(
+        'AppIdbCoreModule is already loaded. Import it in the AppModule only');
```
Comment (szenzaro) [inline src/app/app-idb.module.ts:17]: Fix parenthesis

File: src/app/app-idb.module.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+@NgModule({
+  declarations: [],
+  imports: [],
+  providers: [
+    IdbService,
+  ],
+  bootstrap: [],
+})
+export class AppIdbCoreModule {
+  constructor(@Optional() @SkipSelf() parentModule: AppIdbCoreModule) {
```
Comment (szenzaro) [inline src/app/app-idb.module.ts:14]: Why this? Check if this may have become obsolete.

**Review by szenzaro — CHANGES_REQUESTED** [review]

File: src/app/models/evt-models.ts (hunk 2 above)
Comment (giacomocerre) [reply src/app/models/evt-models.ts:269]: :+1:

File: src/app/app-idb.module.ts (hunk 3 above)
Comment (giacomocerre) [reply src/app/app-idb.module.ts]: :+1:

File: src/app/services/idb.service.ts (hunk 1 above)
Comment (giacomocerre) [reply src/app/services/idb.service.ts]: I remove them and I use the functions of the DB directly.

**Review by ChiaraDipi — APPROVED** [review]

**Review by szenzaro — APPROVED** [review]

**Pull request merged**